src/Evolution/Systems/Ccz4/Ricci.cpp

```diff
@@ -0,0 +1,141 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/Ccz4/Ricci.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Ccz4 {
+template <size_t Dim, typename Frame, typename DataType>
+void ricci(
+    const gsl::not_null<tnsr::ii<DataType, Dim, Frame>*> result,
+    const tnsr::Ijj<DataType, Dim, Frame>& christoffel_second_kind,
+    const tnsr::iJkk<DataType, Dim, Frame>& d_conformal_christoffel_second_kind,
+    const tnsr::ii<DataType, Dim, Frame>& conformal_spatial_metric,
+    const tnsr::II<DataType, Dim, Frame>& inverse_conformal_spatial_metric,
+    const tnsr::ijj<DataType, Dim, Frame>& field_d,
+    const tnsr::iJJ<DataType, Dim, Frame>& field_d_up,
+    const tnsr::i<DataType, Dim, Frame>& field_p,
+    const tnsr::ij<DataType, Dim, Frame>& d_field_p) noexcept {
+  destructive_resize_components(result,
+                                get_size(get<0, 0>(conformal_spatial_metric)));
+  for (auto& component : (*result)) {
+    component = 0.0;
+  }
+
+  for (size_t i = 0; i < Dim; i++) {
+    for (size_t j = i; j < Dim; j++) {
+      for (size_t m = 0; m < Dim; m++) {
+        result->get(i, j) +=
+            d_conformal_christoffel_second_kind.get(m, m, i, j) -
+            d_conformal_christoffel_second_kind.get(j, m, i, m);
+        for (size_t l = 0; l < Dim; l++) {
+          result->get(i, j) +=
```

Comment:
until the docs renders and I can compare to your math there (and verify your math in the docs) I can't really review this. It's a lot of terms. Could you please:
1. document the `Ccz4` namespace `Evolution/Systems/Ccz4/System.hpp` in a separate commit
2. document the actual equation you've coded in addition to what you have
3. add code comments about what the terms are matching to in the documentation

Reply:
In the documentation, I forgot to substitute `m` in for `k` in:
```
 * \f{align}
 *     R_{ij} &=
 *       \partial_k \Gamma^m{}_{ij} - \partial_j \Gamma^m{}_{ik} +
 *       \Gamma^l{}_{ij} \Gamma^m{}_{lk} - \Gamma^l{}_{ik} \Gamma^m{}_{lj}
 * \f}
```

Outside of this definition (once fixed) and the accompanying definition I provided below it for `\partial_k \Gamma^m{}_{ij}` (eq. 18, used for the first two terms in the eq. above), when you say "document the actual equation you've coded in addition to what you have," do you mean that you want me to also write out what `R_{ij}` would be with the all of the terms from eq. 18 substituted (twice) for the first two terms? I originally didn't do this because I thought showing the full substitution might be overkill, so I want to make sure I'm interpreting your request correctly.

Reply:
Yes, I think it would be good to write it out. Someone 5 years from now will look at this code and the more help you can give them the better :)